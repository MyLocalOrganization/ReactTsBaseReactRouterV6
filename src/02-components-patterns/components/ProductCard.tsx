import { createContext } from 'react';
import styles from '../styles/styles.module.css';
import { UseProduct } from '../hooks/usePeoduct';
import { ProductContextProps, ProductCarsProps } from '../interfaces/interfaces';



//contexto global de funciones generales
export const ProductContext = createContext({} as ProductContextProps);

//podria desestructurar el providers
//const { Provider } = ProductContext;

export const ProductCard = ({ children, product }: ProductCarsProps) => {
    //custom hook de funciones 
    const { onDecrement, counter } = UseProduct();

    return (
        <ProductContext.Provider value={{
            onDecrement,
            counter,
            product
        }}>
            <div className={styles.productCard}>
                {children}
            </div>
        </ProductContext.Provider>

    )
}
