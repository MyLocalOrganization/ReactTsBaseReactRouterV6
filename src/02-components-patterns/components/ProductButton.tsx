import { useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from '../styles/styles.module.css';

export const ProductButton = () => {
    //acceso al contexto del padre ProductContext 
    const { onDecrement, counter } = useContext(ProductContext);

    return (
        <div className={styles.buttonsContainer}>
            <button
                className={styles.buttonMinus}
                onClick={() => onDecrement(-1)}
            >-</button>
            <div className={styles.countLabel}>{counter}</div>
            <button
                className={styles.buttonAdd}
                onClick={() => onDecrement(+1)}
            >+</button>
        </div>
    )
}