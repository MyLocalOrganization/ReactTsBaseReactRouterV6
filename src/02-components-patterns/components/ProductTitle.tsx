import { useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from '../styles/styles.module.css';

export const ProductTitle = ({ title }: { title?: string }) => {
    //acceso al contexto del padre ProductContext
    const { product } = useContext(ProductContext);
    if (!title) {
        title = product.title
    }

    return (
        <span className={styles.productDescription}>{title}</span>
    )
}