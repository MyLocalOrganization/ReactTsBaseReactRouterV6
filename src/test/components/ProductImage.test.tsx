import React from 'react'
import render from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../02-components-patterns/components';
import { products1, products2 } from '../products';

test('debe de mostrar el componente con la imagen', () => {
    const wrapper = render.create(
        <ProductCard product={products2}>
            {
                () => (
                    <ProductImage img={products2.img}  />
                )
            }
        </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('debe de mostrar el componente con la imagen del producto', () => {
    const wrapper = render.create(
        <ProductCard product={products1}>
            {
                () => (
                    <ProductImage />
                )
            }
        </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
});
