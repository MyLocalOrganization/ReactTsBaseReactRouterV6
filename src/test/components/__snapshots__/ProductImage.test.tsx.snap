// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe de mostrar el componente con la imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Image"
    className="productImg undefined"
    src="./coffee-mug2.png"
  />
</div>
`;

exports[`debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
